esphome/components/ufire_ec/__init__.py

```diff
@@ -0,0 +1,93 @@
+import esphome.codegen as cg
+from esphome import automation
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_ID,
+    CONF_EC,
+    CONF_TEMPERATURE,
+    DEVICE_CLASS_EMPTY,
+    DEVICE_CLASS_TEMPERATURE,
+    ICON_EMPTY,
+    UNIT_CELSIUS,
+    UNIT_MILLISIEMENS_PER_CENTIMETER,
+)
+
+DEPENDENCIES = ["i2c", "sensor"]
+CODEOWNERS = ["@pvizeli"]
+
+CONF_SOLUTION = "solution"
+CONF_TEMPERATURE_SENSOR = "temperature_sensor"
+CONF_TEMPERATURE_COMPENSATION = "temperature_compensation"
+
+ufire_ec_ns = cg.esphome_ns.namespace("ufire_ec")
+UFireECComponent = ufire_ec_ns.class_(
+    "UFireECComponent", cg.PollingComponent, i2c.I2CDevice
+)
+
+# Actions
+UFireISECalibrateProbeAction = ufire_ec_ns.class_(
+    "UFireISECalibrateProbeAction", automation.Action
+)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(UFireECComponent),
+            cv.Optional(CONF_TEMPERATURE): sensor.sensor_schema(
+                UNIT_CELSIUS, ICON_EMPTY, 1, DEVICE_CLASS_TEMPERATURE
+            ),
+            cv.Optional(CONF_EC): sensor.sensor_schema(
+                UNIT_MILLISIEMENS_PER_CENTIMETER, ICON_EMPTY, 1, DEVICE_CLASS_EMPTY
+            ),
+            cv.Optional(CONF_TEMPERATURE_SENSOR): cv.use_id(sensor.Sensor),
+            cv.Optional(CONF_TEMPERATURE_COMPENSATION, default=25.0): cv.float_,
```

Comment:
There is a helper to validate temperatures correctly in yaml
```suggestion
            cv.Optional(CONF_TEMPERATURE_COMPENSATION, default=25.0): cv.temperature,
```